### PR TITLE
Resource sharing docs

### PR DESCRIPTION
**Pull request by mLavacca**
# Description

Documentation for the resource sharing

### CODE REVIEW

File: docs/design/resource_sharing.md [hunk 1]
```diff
@@ -1,44 +1,93 @@
...
 
-### Remote Reflection
+The VK is able to work in a multi-namespaced environment: whether a new pod belonging to a specific namespace is
+scheduled to the Virtual Node, the VK looks-up for the related translation entry in a resource of type
+`NamespaceNattingTable` associated to the current VK instance. The `NamespaceNattingTable` CRD keeps a translation table
+between local and remote namespaces.
+If the lookup operation doesn't return a valid translation for the local namespace, it creates a new translation entry
+by using the following syntax: `<local-namespace>-<home-cluster-id>`, then creates the remote namespace and starts
+reflecting all the local resources (see below). Once a translation entry for the current namespace is set up, and the
+reflection mechanism is configured, the received pods are sent to the foreign clusters, and their lifecycle is handled
```
Comment (fraborg) [inline docs/design/resource_sharing.md:41]:
```suggestion
reflection mechanism is configured, the received pods are sent to the foreign cluster, and their lifecycle is handled 
```

File: docs/design/resource_sharing.md [hunk 2]
```diff
@@ -1,44 +1,93 @@
...
+Thanks to the `NamespaceNattingTable` resource, the pods statuses reconciliation and the resource reflection are tolerant
+to the VK fault: if the VK pod faults (or is deleted), at the startup of the new VK pod (handled by the replica
+controller), the remote cluster status is kept aligned with the local one seamlessly, thanks to the VK ability to hook
+the previous state and restore its internal representation of both the remote and local resource statuses.
+
+### Limitations
+
+The main limitation of the VK is the reflection of the endpoints, a resource very hard to cope with, due to its
+non-modular nature.
+The problem of the endpoints reflection is based on the resource status ephemeral depending on its high mutation rate
```
Comment (fraborg) [inline docs/design/resource_sharing.md:64]: this phrase is not so clear to me

Comment (mLavacca) [reply docs/design/resource_sharing.md]: Is it clearer now?

**Pull request merged**